internal/util/k8s/client.go

```diff
@@ -47,6 +48,7 @@ func NewK8sClient() (*kubernetes.Clientset, error) {
 			return nil, fmt.Errorf("failed to get cluster config: %w", err)
 		}
 	}
+	cfg.ContentType = runtime.ContentTypeProtobuf
```

Comment:
there is a PV [controller](https://github.com/ceph/ceph-csi/tree/devel/internal/controller), can you please if we need any changes to that as well?

Reply:
Updated the code for that controller.